server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/MIMEMessageConverter.java

```diff
@@ -137,18 +143,25 @@ private void buildMimeHeaders(Message.Builder messageBuilder, CreationMessageId
 
         // note that date conversion probably lose milliseconds!
         messageBuilder.setDate(Date.from(newMessage.getDate().toInstant()), TimeZone.getTimeZone(newMessage.getDate().getZone()));
-        newMessage.getInReplyToMessageId().ifPresent(addInReplyToHeader(messageBuilder::addField));
+        newMessage.getInReplyToMessageId()
+            .ifPresent(id -> addHeader(messageBuilder, IN_REPLY_TO_HEADER, id));
         if (!isMultipart(newMessage, messageAttachments)) {
             newMessage.getHtmlBody().ifPresent(x -> messageBuilder.setContentType(HTML_MEDIA_TYPE, UTF_8_CHARSET));
         }
+        newMessage.getHeaders().entrySet().stream()
+            .filter(header -> ! COMPUTED_HEADERS.contains(header.getKey()))
```

Comment:
Case variations? What if I specify "FROM"? We would be vulnerable as, as far as I remember, header names are not case sensitive.

Reply:
What happens upon empty header names?

Reply:
empty header names should not be supported regarding rfc822